### PR TITLE
Refactor foxy to galactic

### PR DESCRIPTION
**Pull request by yameatmeyourdead**
The day has come.
Update from ROS2 Foxy -> Galactic

Modified the rov and basestation dockerfiles to be based on commonv8_3
Commonv8_3 is 3rd revision of common docker image for armv8

Commonv8_3 has room to expand common apt-packages and pip-packages without having to rebuild 99% of the image

### CODE REVIEW

File: Dockerfile.basestation [hunk 1]
```diff
@@ -32,14 +30,16 @@ RUN udevadm trigger
 
 COPY src .
 
-RUN . /opt/ros/$ROS_DISTRO/install/setup.sh && \
+# build all packages defined in basestation_launch file
+RUN export PYTHONWARNINGS=ignore:::setuptools.command.install,ignore:::setuptools.command.easy_install,ignore:::pkg_resources && \
```
Comment (Hermanoid) [inline Dockerfile.basestation:34]: Ah yes, the ignore statement

File: Dockerfile.commonv8_3 [hunk 2]
```diff
@@ -217,23 +190,56 @@ RUN apt-get update && \
...
-    cmake -DBUILD_SHARED_LIBS=ON .. && \
+    cmake -DBUILD_SHARED_LIBS=ON -DYAML_CPP_INSTALL=ON .. && \
     make -j$(nproc) && \
-    cp libyaml-cpp.so.0.6.0 /usr/lib/aarch64-linux-gnu/ && \
-    ln -s /usr/lib/aarch64-linux-gnu/libyaml-cpp.so.0.6.0 /usr/lib/aarch64-linux-gnu/libyaml-cpp.so.0.6
+    make install
+    #cp libyaml-cpp.so.0.6.0 /usr/lib/aarch64-linux-gnu/ && \
+    #ln -s /usr/lib/aarch64-linux-gnu/libyaml-cpp.so.0.6.0 /usr/lib/aarch64-linux-gnu/libyaml-cpp.so.0.6
 
-### ROS
```
Comment (Hermanoid) [inline Dockerfile.commonv8_3:232]: This sounds like it was fun to figure out

**Review by Hermanoid — COMMENTED** [review]
It is good

File: Dockerfile.basestation (hunk 1 above)
Comment (yameatmeyourdead) [reply Dockerfile.basestation:34]: colcon uses a deprecated function to install our python packages
three possible remedies:
fix colcon
change to c++
or ignore python's complaining

I chose to gag python

File: Dockerfile.commonv8_3 (hunk 2 above)
Comment (yameatmeyourdead) [reply Dockerfile.commonv8_3:232]: The old "steal" from dusty-nv's jetson containers

**Pull request merged**